Image build/release: do not save image for builds; save only for release

BlaineEXE opened the pull request:
**Description of your changes:**

Add an option `SKIP_IMAGE_SAVE` to the Ceph image makefiles with the
intent that a developer can set this variable to `1` or some other
nonzero value to skip the lengthy image saving part of the Rook build.
This will allow Ceph developers to iterate more quickly on dev builds.

Signed-off-by: Blaine Gardner <blaine.gardner@suse.com>

**Which issue is resolved by this Pull Request:**
N/A

**Checklist:**
- [x] Documentation has been updated, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [x] `make vendor` does not cause changes.
- [x] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md#comments)

bassam commented:
Thanks @BlaineEXE. I was thinking about this one too and hoping that we move the image save to the release makefiles instead. Somewhere here: https://github.com/rook/rook/blob/master/build/release/Makefile#L155

what do you think?

BlaineEXE commented:
@bassam, I added the image save steps to the `build/release` Makefile instead and removed the save feature from all the image builds. I simplified the logic a bit by saving every time, as the if statements are complicated, and the image save is likely to be needed for every release build, making them of little use.

As a note, there is currently the assumption that only `linux_<arch>` builds are done, which is true as far as I can tell; I see this assumption throughout a lot of the make code.

bassam commented on images/ceph-toolbox/Makefile:
```diff
@@ -46,10 +46,4 @@ do.build: toolbox-base-image
 	@cp -a . $(TEMP)
 	@cd $(TEMP) && $(SED_CMD) 's|BASEIMAGE|$(TOOLBOX_BASE_IMAGE)|g' Dockerfile
 	@docker build $(BUILD_ARGS) -t $(TOOLBOX_IMAGE) $(TEMP)
-	@[ "$(OLD_IMAGE_ID)" != "$(CURRENT_IMAGE_ID)" ] && [ -n "$(OLD_IMAGE_ID)" ] && docker rmi $(OLD_IMAGE_ID) || true
```
I think you can remove OLD_IMAGE_ID and CURRENT_IMAGE_ID above too.

BlaineEXE replied:
Good catch. My most recent push should fix all these.

bassam commented on build/release/Makefile:
```diff
@@ -153,7 +153,11 @@ promote.output:
+	@docker tag $(BUILD_REGISTRY)/$(2)-$(3) $(1)/$(2)-$(3):$(VERSION)
+	@# Save image as _output/images/linux_<arch>/<image>.tar.gz (no builds for darwin or windows)
+	@mkdir -p $(OUTPUT_DIR)/images/linux_$(3)
+	@docker save $(BUILD_REGISTRY)/$(2)-$(3) | gzip -c > $(OUTPUT_DIR)/images/linux_$(3)/$(2).tar.gz
```
might be better to change IMAGE_ARCHS above to not filter out the leading `linux_` and change the docker save here to just use `$(OUTPUT_DIR)/images/$(3)/$(2).tar.gz`

BlaineEXE replied:
I had considered that. As long as it won't cause issues elsewhere, I'll get on it.

BlaineEXE commented:
@bassam Care to merge? Thanks for the feedback. =)
